### PR TITLE
feat: [SFEQS-1496] Add lollipop middleware to `io-sign`

### PR DESCRIPTION
**Pull request by grausof**
This PR adds lollipop middleware to `io-sign`, specifically to the `createSignature` endpoint.

#### List of Changes
- Added lollipop headers to `createSignature` endpoint.
- Added lollipop middleware to `createSignature` endpoint.
- Updated openapi specification
- Updated unit-test

#### Motivation and Context
`io-sign` needs to use lollipop to create signatures!

#### How Has This Been Tested?
Tested locally

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: src/controllers/ioSignController.ts [hunk 1]
```diff
@@ -137,21 +167,22 @@ export default class IoSignController {
...
     | IResponseSuccessJson<SignatureDetailView>
   > =>
-    withUserFromRequest(req, async user =>
+    withUserFromRequest(req, user =>
       pipe(
-        req.body,
-        CreateSignatureBody.decode,
-        E.mapLeft(responseErrorValidation),
+        locals,
+        withLollipopLocals,
```
Comment (gunzip) [inline src/controllers/ioSignController.ts:182]: can you add a comment on what's happening here?
(what are "lollipop locals"?)

File: src/controllers/ioSignController.ts [hunk 2]
```diff
@@ -167,13 +198,26 @@ export default class IoSignController {
...
               )
             }),
-            TE.map(({ userProfile, signerId }) =>
+            TE.chainW(({ signerId, userProfile }) =>
+              pipe(
+                req.body,
+                CreateSignatureBody.decode,
+                E.mapLeft(responseErrorValidation),
+                TE.fromEither,
+                TE.map(body => ({
```
Comment (gunzip) [inline src/controllers/ioSignController.ts:207]:
```suggestion
                TE.map(signatureBody => ({
```

File: src/controllers/ioSignController.ts (hunk 1 above)
Comment (grausof) [reply src/controllers/ioSignController.ts:182]: `withLollipopLocals` is an utility function of lollipop domain. Its role is described in the [comment](https://github.com/pagopa/io-backend/blob/8a92889e92d71877e7f4a6d29b77e42899828e93/src/types/lollipop.ts#L54) of the function.

Comment (gunzip) [reply src/controllers/ioSignController.ts:182]: it's still hard to figure it out what's happening here despite that comment

Comment (grausof) [reply src/controllers/ioSignController.ts:182]: I add a comment!

**Review by gunzip — APPROVED** [review]

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**